cmd/genallocs/genesis_alloc.go

```diff
@@ -0,0 +1,189 @@
+package genallocs
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math/big"
+	"os"
+	"path/filepath"
+
+	"github.com/dominant-strategies/go-quai/common"
+	"github.com/dominant-strategies/go-quai/params"
+	orderedmap "github.com/wk8/go-ordered-map/v2"
+	"lukechampine.com/blake3"
+)
+
+// GenesisAccount is an account in the state of the genesis block.
+type GenesisAccount struct {
+	UnlockSchedule  int                                      `json:"unlockSchedule"`
+	Address         common.Address                           `json:"address"`
+	Award           *big.Int                                 `json:"award"`
+	Vested          *big.Int                                 `json:"vested"`
+	LumpSumMonth    uint64                                   `json:"lumpSumMonth"`
+	BalanceSchedule *orderedmap.OrderedMap[uint64, *big.Int] `json:"balanceSchedule"` // Map of blockNumber->balanceUnlocked (at that block).
+}
+
+// Defines the unlockSchedule parameters.
+// Some unlockSchedules have a lumpSum payment, some at TGE, some at 1 year.
+// All unlockSchedules begin regular unlocks after 1 year.
+type unlockSchedule struct {
+	unlockDuration    uint64 // Total unlocking duration in months. The first year cliff is not part of unlocking.
```

Comment:
[nit] The comment here is confusing. Might be better to say `// Number of monthly unlocks after the lump sum`